compute/ip_address_associations_test.go

```diff
@@ -0,0 +1,163 @@
+package compute
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+	"testing"
+
+	"github.com/hashicorp/go-oracle-terraform/helper"
+	"github.com/hashicorp/go-oracle-terraform/opc"
+)
+
+const (
+	_IPAddressAssociationTestName        = "test-acc-ip-address-association"
+	_IPAddressAssociationTestDescription = "testing ip address association"
+)
+
+func TestAccIPAddressAssociationsLifeCycle(t *testing.T) {
+	helper.Test(t, helper.TestCase{})
+
+	iClient, nClient, vnClient, err := getVirtNICsTestClients()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ipNetwork, err := createTestIPNetwork(nClient, _IPNetworkTestPrefix)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer destroyIPNetwork(t, nClient, ipNetwork.Name)
+
+	// In order to get details on a Virtual NIC we need to create the following resources
+	// - IP Network
+	// - Instance
+
+	instanceInput := &CreateInstanceInput{
+		Name:      _VirtNicInstanceTestName,
+		Label:     _VirtNicInstanceTestLabel,
+		Shape:     _VirtNicInstanceTestShape,
+		ImageList: _VirtNicInstanceTestImage,
+		Networking: map[string]NetworkingInfo{
+			"eth0": {
+				IPNetwork: ipNetwork.Name,
+				Vnic:      "eth0",
+			},
+		},
+	}
+
+	createdInstance, err := iClient.CreateInstance(instanceInput)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tearDownInstances(t, iClient, createdInstance.Name, createdInstance.ID)
+
+	// Use the static "eth0" interface, as we statically created that above
+	createdVNIC := createdInstance.Networking["eth0"].Vnic
+	getVNICInput := &GetVirtualNICInput{
+		Name: createdVNIC,
+	}
+
+	vNIC, err := vnClient.GetVirtualNIC(getVNICInput)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ipaClient, err := getIPAddressReservationsTestClients()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	resName := fmt.Sprintf("%s-%d", _TestIPAddressResName, helper.RInt())
+
+	input := &CreateIPAddressReservationInput{
+		Description:   _TestIPAddressResDesc,
+		IPAddressPool: PrivateIPAddressPool,
+		Name:          resName,
+		Tags:          []string{_TestIPAddressResTag},
+	}
+
+	ipRes, err := ipaClient.CreateIPAddressReservation(input)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer destroyIPAddressReservation(t, ipaClient, resName)
+
+	svc, err := getIPAddressAssociationsClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	createInput := &CreateIPAddressAssociationInput{
+		Name:                 _IPAddressAssociationTestName,
+		Description:          _IPAddressAssociationTestDescription,
+		IPAddressReservation: ipRes.Name,
+		VNIC:                 vNIC.Name,
+		Tags:                 []string{"testing"},
+	}
+
+	createdIPAddressAssociation, err := svc.CreateIPAddressAssociation(createInput)
+	if err != nil {
+		t.Fatal(err)
+	}
+	log.Print("IP Address Association succcessfully created")
+	defer destroyIPAddressAssociation(t, svc, _IPAddressAssociationTestName)
```

Comment:
Just for safety reasons, can throw the defer before the `log`. Highly doubt `log.Print()` will fail, but stranger things have happened.